src/com/griddynamics/cinema/CinemaCli.java

```diff
@@ -0,0 +1,134 @@
+package com.griddynamics.cinema;
+
+import java.util.Scanner;
+
+public class CinemaCli {
+    
+    private static final int ONE_HUNDRED_PERCENT = 100;
+    
+    private static final int LARGE_CINEMA_HALL_MINIMUM_SEATS_AMOUNT = 60;
+    private static final int TICKET_FULL_PRICE = 10;
+    private static final int TICKET_DISCOUNT_PRICE = 8;
+    
+    private final Scanner scanner = new Scanner(System.in);
+    
+    private final boolean isLargeCinemaHall;
+    
+    private final int amountOfRows;
+    private final int amountOfSeatsInEachRow;
+    private final int totalSeatsAmount;
+    
+    private final int amountOfRowsWithFullPriceSeats;
+    private final int amountOfRowsWithDiscountPriceSeats;
+    
+    private final Cinema cinema;
+    
+    public CinemaCli(final Cinema cinema) {
+        this.cinema = cinema;
+        
+        amountOfRows = this.cinema.getHall().getRows().size();
+        amountOfSeatsInEachRow = this.cinema.getHall().getRows().get(0).getSeatsInEachRow().size();
+        
+        totalSeatsAmount = amountOfRows * amountOfSeatsInEachRow;
+        
+        amountOfRowsWithFullPriceSeats = amountOfRows / 2;
+        amountOfRowsWithDiscountPriceSeats = (amountOfRows + 1) / 2;
+        
+        isLargeCinemaHall = totalSeatsAmount >= LARGE_CINEMA_HALL_MINIMUM_SEATS_AMOUNT;
+    }
+    
+    public Cinema getCinema() {
+        return cinema;
+    }
+    
+    public void interact() {
+        int userOption = 0;
+        
+        do {
+            System.out.print("\n1. Show the seats\n2. Buy a ticket\n3. Statistics\n0. Exit\n> ");
+            try {
+                userOption = scanner.nextInt();
+                if (userOption < 0 || userOption > 3) {
+                    throw new RuntimeException();
+                }
+                switch (userOption) {
+                    case 1 -> showSeats();
+                    case 2 -> ticketPurchasingCLI();
+                    case 3 -> statistics();
+                }
+            } catch (RuntimeException e) {
+                System.out.println("\nInvalid input data. Please retry.");
+            }
+        } while (userOption != 0);
+    }
+    
+    private void ticketPurchasingCLI() throws RuntimeException {
+        int row;
```

Comment:
RuntimeException and its subclasses are unchecked exceptions. Unchecked exceptions do not need to be declared in a method or constructor's throws clause if they can be thrown by the execution of the method or constructor and propagate outside the method or constructor boundary.

Reply:
Got it!